### PR TITLE
Bugfix

### PR DESCRIPTION
**Pull request by wendig0x**
Some performance fixes and check return value of StringFromGUID2

### CODE REVIEW

**Comment by andreas-becker** [comment]
@wendig0x, please fix the conflicts.

**Comment by wendig0x** [comment]
@andreas-becker done

**Review by andreas-becker — APPROVED** [review]

**Comment by wendig0x** [comment]
@alt3r-3go Thanks for the great code review! I'm a little busy at the moment, I'll add comments and corrections later.

File: src/Common/BootEncryption.cpp [hunk 1]
```diff
@@ -5421,11 +5424,8 @@ namespace VeraCrypt
 		}
 
 		// Change the PKCS-5 PRF if requested by user
-		if (pkcs5 != 0)
-		{
-			cryptoInfo->pkcs5 = pkcs5;
-			RandSetHashFunction (pkcs5);
-		}
+		cryptoInfo->pkcs5 = pkcs5;
+		RandSetHashFunction (pkcs5);
```
[the remark this answers is not in the document]
Comment (wendig0x) [reply src/Common/BootEncryption.cpp:5428]: This check is in the code above: if (pkcs5 != 0) throw ParameterIncorrect (SRC_POS);

File: src/Common/Dlgcode.c [hunk 2]
```diff
@@ -15262,7 +15268,7 @@ DWORD PasswordEditDropTarget::GotEnter(void)
 	HWND hChild = WindowFromPoint (m_DropPoint);
 	// check that we are on password edit control (we use maximum length to correctly identify password fields since they don't always have ES_PASSWORD style (if the the user checked show password)
 	if (hChild && GetClassName (hChild, szClassName, ARRAYSIZE (szClassName)) && (0 == _tcsicmp (szClassName, _T("EDIT")))
-		&& (dwStyles = GetWindowLong (hChild, GWL_STYLE)) && !(dwStyles & ES_NUMBER)
+		&& (dwStyles = GetWindowLongPtr (hChild, GWL_STYLE)) && !(dwStyles & ES_NUMBER)
```
[the remark this answers is not in the document]
Comment (wendig0x) [reply src/Common/Dlgcode.c]: fixed

File: src/Common/Dlgcode.c [hunk 3]
```diff
@@ -14547,7 +14547,10 @@ static bool RunAsDesktopUser(
 		LookupPrivilegeValueW(NULL, SE_INCREASE_QUOTA_NAME, &tkp.Privileges[0].Luid);
 		tkp.Privileges[0].Attributes = SE_PRIVILEGE_ENABLED;
 
-		SetThreadToken (NULL, NULL);
+		if (!SetThreadToken(NULL, NULL))
+		{
+			goto cleanup;
+		}
```
[the remark this answers is not in the document]
Comment (wendig0x) [reply src/Common/Dlgcode.c]: fixed

**Comment by DLL125** [comment]
@wendig0x 
Are you still working on these?

**Pull request closed without merging**